File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        f,s=head,head
        for _ in range(n): f=f.next      
        if not f: return head.next
        while f.next: f,s = f.next,s.next
        s.next=s.next.next
        return head

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if(!head || !head->next)return nullptr;
        ListNode *s=head,*f=head;
        while(n--){
            f=f->next;
        }
        if(!f)return head->next;
        while(f->next){
            f=f->next;
            s=s->next;
        }
        s->next=s->next->next;
        return head;
    }
};